### PR TITLE
Use default build env and test config for test times

### PR DESCRIPTION
**Pull request by clee2000**
Redo of #107312 

Pairs with https://github.com/pytorch/test-infra/pull/4476

If build env and test config combo cannot be found in the test times, use default.  Then we don't have to go manually change the test-times.json a new job is added or we update the jobs.

### CODE REVIEW

File: test/run_test.py [hunk 1]
```diff
@@ -1416,21 +1416,30 @@ def get_selected_tests(options) -> List[ShardedTest]:
...
+
+    with open(path) as f:
+        test_times_file = cast(Dict[str, Any], json.load(f))
+    build_environment = os.environ.get("BUILD_ENVIRONMENT")
     test_config = os.environ.get("TEST_CONFIG")
-    if test_config not in test_file_times:
+    if test_config in test_times_file.get(build_environment, {}):
+        print("Found test times from artifacts")
+        return test_times_file[build_environment][test_config]
+    elif test_config in test_times_file["default"]:
```
Comment (huydhn) [inline test/run_test.py:1431]: Nit: you could use the same `test_times_file.get("default", {})` here I think

**Review by huydhn — APPROVED** [review]
LGTM!

File: test/run_test.py (hunk 1 above)
Comment (clee2000) [reply test/run_test.py:1431]: default should be guaranteed to always be there

**Pull request closed without merging**